packages/react-dom/src/__tests__/ReactErrorBoundaries-test.internal.js

```diff
@@ -576,103 +578,148 @@ describe('ReactErrorBoundaries', () => {
+  it('does not swallow exceptions on unmounting without boundaries', async () => {
     const container = document.createElement('div');
-    ReactDOM.render(<BrokenComponentWillUnmount />, container);
-    expect(() => {
-      ReactDOM.unmountComponentAtNode(container);
-    }).toThrow('Hello');
+    const root = ReactDOMClient.createRoot(container);
+    await act(async () => {
+      root.render(<BrokenComponentWillUnmount />);
+    });
+    await expect(async () => {
+      await act(async () => {
+        root.unmount();
+      });
+    }).rejects.toThrow('Hello');
   });
 
-  it('prevents errors from leaking into other roots', () => {
+  it('prevents errors from leaking into other roots', async () => {
     const container1 = document.createElement('div');
+    const root1 = ReactDOMClient.createRoot(container1);
     const container2 = document.createElement('div');
+    const root2 = ReactDOMClient.createRoot(container2);
     const container3 = document.createElement('div');
+    const root3 = ReactDOMClient.createRoot(container3);
 
-    ReactDOM.render(<span>Before 1</span>, container1);
-    expect(() => {
-      ReactDOM.render(<BrokenRender />, container2);
-    }).toThrow('Hello');
-    ReactDOM.render(
-      <ErrorBoundary>
-        <BrokenRender />
-      </ErrorBoundary>,
-      container3,
-    );
+    await act(async () => {
+      root1.render(<span>Before 1</span>);
+    });
+    await expect(async () => {
+      await act(async () => {
+        root2.render(<BrokenRender />);
+      });
+    }).rejects.toThrow('Hello');
+    await act(async () => {
+      root3.render(
+        <ErrorBoundary>
+          <BrokenRender />
+        </ErrorBoundary>,
+      );
+    });
     expect(container1.firstChild.textContent).toBe('Before 1');
     expect(container2.firstChild).toBe(null);
     expect(container3.firstChild.textContent).toBe('Caught an error: Hello.');
 
-    ReactDOM.render(<span>After 1</span>, container1);
-    ReactDOM.render(<span>After 2</span>, container2);
-    ReactDOM.render(
-      <ErrorBoundary forceRetry={true}>After 3</ErrorBoundary>,
-      container3,
-    );
+    await act(async () => {
+      root1.render(<span>After 1</span>);
+    });
+    await act(async () => {
+      root2.render(<span>After 2</span>);
+    });
+    await act(async () => {
+      root3.render(<ErrorBoundary forceRetry={true}>After 3</ErrorBoundary>);
+    });
     expect(container1.firstChild.textContent).toBe('After 1');
     expect(container2.firstChild.textContent).toBe('After 2');
     expect(container3.firstChild.textContent).toBe('After 3');
 
-    ReactDOM.unmountComponentAtNode(container1);
-    ReactDOM.unmountComponentAtNode(container2);
-    ReactDOM.unmountComponentAtNode(container3);
+    await act(async () => {
+      root1.unmount();
+    });
+    await act(async () => {
+      root2.unmount();
+    });
+    await act(async () => {
+      root3.unmount();
+    });
```

Comment:
```suggestion
    root1.unmount();
    root2.unmount();
    root3.unmount();
```

You don't need act for unmount, it's sync.

Reply:
Fixed everywhere in this file.